src/sentry/integrations/repository/issue_alert.py

```diff
@@ -21,3 +48,33 @@ def __init__(self, logger: Logger) -> None:
     @classmethod
     def default(cls) -> IssueAlertNotificationMessageRepository:
         return cls(logger=_default_logger)
+
+    def get_parent_notification_message(
+        self, rule_id: int, group_id: int, rule_action_uuid: str
+    ) -> IssueAlertNotificationMessage | None:
+        """
+        Returns the parent notification message for a metric rule if it exists, otherwise returns None.
+        Will raise an exception if the query fails and logs the error with associated data.
+        """
+        try:
+            instance: NotificationMessage = self._model.objects.get(
+                rule_fire_history__rule__id=rule_id,
+                rule_fire_history__group__id=group_id,
+                rule_action_uuid=rule_action_uuid,
+                parent_notification_message__isnull=True,
+                error_code__isnull=True,
+            )
+            return IssueAlertNotificationMessage.from_model(instance=instance)
+        except NotificationMessage.DoesNotExist:
+            return None
+        except Exception as e:
+            self._logger.exception(
+                "Failed to get parent notification for issue rule",
+                exc_info=e,
+                extra={
+                    "rule_id": rule_id,
+                    "group_id": group_id,
+                    "rule_action_uuid": rule_action_uuid,
+                },
+            )
+            raise
```

Comment:
How is this being handled in the end?

Reply:
It's up to the caller to decide how they want to proceed, and if this is a blocking data piece for them or not. For our use case in threads notification, we'll probably block, or fallback to just posting the message as a normal message in the channel (as apposed to a thread)

Reply:
This is something we can get clarity on from product